### ISSUE TITLE
Endpoint az sequence inconsistent with input subnet

### ISSUE THREAD

**Issue by latelan** [mattyait/terraform-aws-network-firewall#11]
There is a scenario with three azs and two groups of subnets, one group for NAT GW subnet and another group for FW subnet. Each subnet has a routing table. In the NAT GW routing table, there is a route pointing to the FW endpoint in the same az as follow.
<img width="515" alt="image" src="https://user-images.githubusercontent.com/4263137/219931625-c954fe9d-f555-414d-95b1-87023c6397e3.png">

and use the code mentioned at https://github.com/mattyait/terraform-aws-network-firewall/pull/10, while I found that some routes are incrorrect, such as a route which in az A route table points to the az B endpoint.

At first, I thought the order of input subnets is disrupted by [function `toset()`](https://github.com/mattyait/terraform-aws-network-firewall/blob/master/main.tf#L10-L16) , causing the endpoint sequence to become disordered. So I tested without function `toset()`, but it's didn't work.

In summary, How to ensure the consistency of input subnets and enpoints order in AZ?

**Comment by fortunecookiezen** [comment]
I did not find a way to ensure consistency of order of output of the endpoints so I opted to create a map of the output and perform a lookup to get the correct route table to endpoint mappings:
https://github.com/fortunecookiezen/aws-tf-egress-vpc-tgw-nwf/blob/main/locals.tf
```
locals {
  vpce_lookup = { for state in aws_networkfirewall_firewall.egress.firewall_status[0].sync_states : state.availability_zone => state.attachment[0].endpoint_id }
}
```
Then I could lookup the endpoint via availability zone and associate it with the correct vpc endpoint id:
https://github.com/fortunecookiezen/aws-tf-egress-vpc-tgw-nwf/blob/main/main.tf#L600
```
resource "aws_route" "public_home_net" {
  count = length(var.public_subnets)

  route_table_id         = element(aws_route_table.public_subnet.*.id, count.index)
  destination_cidr_block = var.home_net
  vpc_endpoint_id        = lookup(local.vpce_lookup, var.availability_zone_names[count.index])

  timeouts {
    create = "5m"
  }
}
```

**Comment by latelan** [comment]
@fortunecookiezen good idea

**Comment by mattyait** [comment]
@latelan I got what you are trying to do using this module. To achieve the correct assignment of VPC as per AZ as per your code [#10 ](https://github.com/mattyait/terraform-aws-network-firewall/pull/10) I have added a PR [#12](https://github.com/mattyait/terraform-aws-network-firewall/pull/12) that will add few more output value which can be use later to achieve your requirement too.

After this merge I believe you just need this code which will assign the correct endpoint with correct AZ in your route table.

    resource "aws_route" "central_egress_tgw" {
      count = length(module.central_egress_vpc.public_subnets)
      route_table_id         = element(module.central_egress_vpc.intra_route_table_ids[*], count.index)
      destination_cidr_block = element(["10.10.0.0/28", "10.10.0.16/28", "10.10.0.32/28"],count.index)
      vpc_endpoint_id        = lookup(module.network_firewall.endpoint_id_az, local.azs[count.index])
      depends_on = [module.central_egress_vpc, module.network_firewall]
    }
